continuous-testing/use-cases/markmap.md

```diff
@@ -1,56 +1,134 @@
+                * Using Java
+                    * JProfiler
+                    * JMeter
+                    * JPF
+                * Using Python
+                    * Scalene
+                    * sys.settrace()/inspect/pdb
+                * Using C#
+                    * Performance Profiler
+                    * Parasoft Ensure
+                    * ANTS Profiler
+                * Using C/C++
+                    * Valgrind
+                    * LLVM/Clang
+        * For security
+            * For vulnerability reporting and code checks
+                * Using most languages
+                    * Scrub
+                    * Veracode
     * For Web Applications
-        * Service Endpoints (APIs)
-            * Messaging-based
-                * RESTful Services
-                * SOAP
-                * Tools (JSON/XML)
-                    * Using Apache JMeter
-                    * Using Insomnia
-                    * Using Paw
-                    * Using Postman
-                    * Using REST-Assured (REST, RPC)
-                    * Using SOAPUI
-            * Procedure-based
-                * RPC
-                    * Using Shell/Powershell
-                    * Using PortQry
-        * Web Interfaces (browsers)
-            * Using Cypress
-            * Using Playwright
-            * Using Robot Framework
-            * Using Selenium + Add-on
-                * Cucumber* (Java + others) / Behave (Python) / Specflow (C#)
-                * Watir (Ruby)
-                * Home Grown
-    * For Backend (Load testing?)
-        * Using Chaos Monkey
-        * Using Robot Framework
-        * Using Cucumber Open
-        * For Mobile
+        * For service endpoints (APIs)
+            * For RESTful endpoints
+                * Using API platforms
+                    * Apache JMeter
+                    * Insomnia
+                    * Paw
+                    * Postman
+                    * REST-Assured (REST, RPC)
+                    * SOAPUI
+            * For procedural-based endpoints
+                * Using RPC/remote processing
+                    * Shell/Powershell
+                    * PortQry
+        * For web interfaces
+            * For user interaction
+                * Using browser simulation
+                    * Cypress
+                    * Playwright
+                    * Robot Framework
+                    * Selenium + Cucumber* (Java + others) / Behave (Python) / Specflow (C#)
+                    * Selenium + Watir (Ruby)
+                    * Selenium + Home Grown
+    * For Load testing
+        * For failure profiles and resource analysis
+            * For server deployments
+                * Using request automation
+                    * Apache JMeter
+                    * Chaos Monkey
+                    * Robot Framework
+                    * Cucumber Open
+    * For Mobile
+        * For feature validation
```

Comment:
Do you anticipate alternate category branches here? i.e. "Not with feature validation"?

If you don't plan to have tools that fall under separate "branches" of feature categories, as listed here, then I think you probably don't need to include sub-categories here. There's a bit of overlap here between the Trade Study table and features listed in this use case diagram (which is okay), which is a bit of an art.